### PR TITLE
allow to adjust Nextcloud apps that get installed upon first startup

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/issues/1198

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: Containers/nextcloud/entrypoint.sh [hunk 1]
```diff
@@ -229,12 +229,12 @@ if ! [ -f "$NEXTCLOUD_DATA_DIR/skip.update" ]; then
...
-            php /var/www/html/occ app:install tasks
-            php /var/www/html/occ app:install calendar
-            php /var/www/html/occ app:install contacts
-            php /var/www/html/occ app:install apporder
+            if [ -n "$STARTUP_APPS" ]; then
+                read -ar STARTUP_APPS_ARRAY <<< "$STARTUP_APPS"
+                for app in "${STARTUP_APPS_ARRAY[@]}"; do
+                    php /var/www/html/occ app:install "$app"
+                done
+            fi
```
Comment (Zoey2936) [inline Containers/nextcloud/entrypoint.sh:237]: I don't know if this is a problem on my side, but I don't get this loop running

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh:237]: What exactly did you do? It will only run this at the very first installation of Nextcloud...

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: okay just tested. Indeed it didnt work. Now it should. Feel free to try again :)

Comment (Zoey2936) [reply Containers/nextcloud/entrypoint.sh]: I've exported STARTUP_APPS using different ways:
```
export STARTUP_APPS=twofactor_totp deck tasks calendar contacts apporder
export STARTUP_APPS="twofactor_totp deck tasks calendar contacts apporder"
export STARTUP_APPS=(twofactor_totp deck tasks calendar contacts apporder)
```
And then executed the following script
```bash
#!/bin/bash
if [ -n "$STARTUP_APPS" ]; then
read -ra STARTUP_APPS_ARRAY <<< "$STARTUP_APPS"
for app in "${STARTUP_APPS_ARRAY[@]}"; do
echo "$app"
done fi
```
And I never got any output
grafik

Maybe I do something wrong, but I don't know

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: > export STARTUP_APPS="twofactor_totp deck tasks calendar contacts apporder"

can you try to put that at the beginning of the script and try again?

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: image

Comment (Zoey2936) [reply Containers/nextcloud/entrypoint.sh]: grafik
Made it work

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: Cool!

**Review by Zoey2936 — APPROVED** [review]

**Comment by szaimen** [comment]
Thanks Zoeye! Will merge this as soon as the next stable release is out :)

File: docker-compose.yml [hunk 2]
```diff
@@ -28,6 +28,7 @@ services:
       # - NEXTCLOUD_MAX_TIME=3600 # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-max-execution-time-for-nextcloud
       # - TRUSTED_CACERTS_DIR=/path/to/my/cacerts # CA certificates in this directory will be trusted by the OS of the nexcloud container (Useful e.g. for LDAPS) See See https://github.com/nextcloud/all-in-one#how-to-trust-user-defiend-certification-authorities-ca
       # - COLLABORA_SECCOMP_DISABLED=false # Setting this to true allows to disable Collabora's Seccomp feature. See https://github.com/nextcloud/all-in-one#how-to-disable-collaboras-seccomp-feature
+      # - NEXTCLOUD_STARTUP_APPS=twofactor_totp deck tasks calendar contacts apporder # Allows to modify the Nextcloud apps that are installed on starting AIO the first time
```
Comment (szaimen) [inline docker-compose.yml:31]: we could add a link to the readme here

Comment (szaimen) [reply docker-compose.yml]: done

**Pull request merged**